eclair-core/src/main/scala/fr/acinq/eclair/payment/Bolt11Invoice.scala

```diff
@@ -414,7 +414,7 @@ object Bolt11Invoice {
       .typecase(2, dataCodec(bits).as[UnknownTag2])
       .typecase(3, dataCodec(listOfN(extraHopsLengthCodec, extraHopCodec)).as[RoutingInfo])
       .typecase(4, dataCodec(bits).as[UnknownTag4])
-      .typecase(5, dataCodec(bits).xmap[Features](Features(_), features2bits).as[InvoiceFeatures])
+      .typecase(5, dataCodec(bits).xmap[Features[InvoiceFeature]](Features(_).invoiceFeatures(), features2bits).as[InvoiceFeatures])
```

Comment:
Unfortunately I think this breaks the signature check in `fromString` if the invoice contains a non-invoice feature bit.
We should add test cases:

- read an invoice containing non-invoice known optional feature bits
- read an invoice containing non-invoice known mandatory feature bits
- read an invoice containing unknown optional feature bits
- read an invoice containing unknown mandatory feature bits

We can probably rework the `fromString` method to keep all feature bits for the signature check while still keeping the benefits of using correctly typed features once the checks are done.

Reply:
done